task-queue/src/main/java/org/hawkular/metrics/tasks/api/RepeatingTrigger.java

```diff
@@ -0,0 +1,185 @@
+
+    private long delay;
+
+    private Integer repeatCount;
+
+    private int executionCount;
+
+    public static class Builder {
+
+        private RepeatingTrigger trigger = new RepeatingTrigger();
+
+        public Builder withInterval(int interval, TimeUnit timeUnit) {
+            trigger.interval = TimeUnit.MILLISECONDS.convert(interval, timeUnit);
+            return this;
+        }
+
+        public Builder withDelay(int delay, TimeUnit timeUnit) {
+            trigger.delay = TimeUnit.MILLISECONDS.convert(delay, timeUnit);
+            return this;
+        }
+
+        public Builder withRepeatCount(int count) {
+            trigger.repeatCount = count;
+            return this;
+        }
+
+        public RepeatingTrigger build() {
+            trigger.triggerTime = trigger.getExecutionDateTime().getMillis() + trigger.delay;
+            trigger.executionCount = 1;
+            return trigger;
+        }
+
+    }
+
+    private RepeatingTrigger() {
+    }
+
+    public RepeatingTrigger(long interval) {
+        this.interval = interval;
+    }
+
+    public RepeatingTrigger(long interval, long delay, long triggerTime) {
+        this.interval = interval;
+        this.delay = delay;
+        this.triggerTime = triggerTime;
+    }
+
+    // TODO shoud this constructor be exposed in the client API?
+    // We need this constructor for use by the scheduler when creating a trigger from a row
+    // in the database.
+    public RepeatingTrigger(long interval, long delay, long triggerTime, int executionCount) {
+        this.interval = interval;
+        this.delay = delay;
+        this.triggerTime = triggerTime;
+        this.executionCount = executionCount;
+    }
+
+    public RepeatingTrigger(long interval, long delay, long triggerTime, int repeatCount, int executionCount) {
+        this.interval = interval;
+        this.delay = delay;
+        this.triggerTime = triggerTime;
+        this.executionCount = executionCount;
+        this.repeatCount = repeatCount == 0 ? null : repeatCount;
+        this.executionCount = executionCount;
+    }
+
+    public long getInterval() {
+        return interval;
+    }
+
+    public long getDelay() {
+        return delay;
+    }
+
+    @Override
+    public long getTriggerTime() {
+        return triggerTime;
+    }
+
+    public Integer getRepeatCount() {
+        return repeatCount;
+    }
+
+    public int getExecutionCount() {
+        return executionCount;
+    }
+
+    @Override
+    public Trigger nextTrigger() {
+        if (repeatCount != null && executionCount + 1 > repeatCount) {
+            return null;
+        }
+        RepeatingTrigger next = new RepeatingTrigger();
+        next.interval = interval;
+        next.delay = delay;
+        next.triggerTime = triggerTime + interval;
+        next.repeatCount = repeatCount;
+        next.executionCount = executionCount + 1;
+
+        return next;
+    }
+
+    private DateTime getExecutionDateTime() {
```

Comment:
I'm not sure about making it a default method because it is not part of the public API. It really is just an internal helper method. For now I have introduced an abstract base class and moved the method there to eliminate the duplication. I have also moved RepeatingTrigger.now into AbstractTrigger. It is used for testing with a virtual clock. 

The getExecutionTime method is used in several places. It was basically copied from metrics-core-impl. The task scheduler really should not depend on the core though. Maybe there needs to be some date utils module in hawkular-commons.